RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,226 @@
+| redeem_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `redeem_identity` |
+| secret           | The pre-image to the `secret_hash` used to generate the HTLC                                            |
+| `01`             | A single byte used to activate the redeem path in the `OP_IF`                                           |
+| contract         | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+For how to use this to construct the redeem transaction see [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-id).
+
+To be notified of the redeem event both parties may watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+If Bitcoin is the `beta_ledger`, then the funder (Bob) MUST watch for such a transaction and then extract the `secret` from the witness data and continue the protocol.
+
+#### Refund
+
+To refund the HTLC, the funder should submit a transaction to the blockchain which spends the P2WSH output.
+The funder can use the following witness data to spend the output after the `expiry`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| refund_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `refund_pubkey`    |
+| refund_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `refund_identity` |
+| `00`             | A single byte used to activate the refund path in the `OP_IF`                                           |
+| contract         | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+To be notified of the refund event both parties may watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+
+## Registry extension
+
+This RFC extends the [registry](./registry.md) with an identity definition for the Bitcoin ledger:
+
+| Ledger  | Identity Name | JSON Encoding            | Description                                                                           |
+|:--------|:--------------|:-------------------------|---------------------------------------------------------------------------------------|
+| Bitcoin | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA256 and then RIPEMD160 to a SECP256k1 compressed public key |
+
+# Examples
+
+## Example RFC003 message body
+
+This following shows what a SWAP message looks like where the `alpha_ledger` is Bitcoin, the `alpha_asset` is 1 Bitcoin (with `...` being used where the value is only relevant for the `beta_ledger`).
+
+``` json
```

Comment:
Can we have a complete example here, i.e. without `{...}`?

Reply:
I am not sure if I want that. It would add unnecessary complexity and detail that this RFC is not concerned with. Like, which ledger to choose. Technically, no other ledgers are defined yet. We just added them to the registry.

To me, the example is clear enough without those details.

Reply:
In that particular case it's about 5 additional lines and not more complex :) 
I'm personally always annoyed when finding incomplete examples.

Reply:
I find incomplete examples in *software libraries* annoying. But I think this example wouldn't benefit from being complete as it is just trying to demonstrate how to compile the HTLC. 

There is a complete example in RFC-003. I would be fine with deleting this one in favour of that one (and just having the table and resulting contract and addresses.